packages/webapp/components/invite/Referral.tsx

```diff
@@ -0,0 +1,227 @@
+import { Modal } from '@dailydotdev/shared/src/components/modals/common/Modal';
+import {
+  ModalKind,
+  ModalSize,
+} from '@dailydotdev/shared/src/components/modals/common/types';
+import CloseButton from '@dailydotdev/shared/src/components/CloseButton';
+import { anchorDefaultRel } from '@dailydotdev/shared/src/lib/strings';
+import { JoinPageProps } from './common';
+
+export function Referral({
+  referringUser,
+  redirectTo,
+  token,
+}: JoinPageProps): ReactElement {
+  const router = useRouter();
+  const genericReferral = useFeature(feature.genericReferral);
+  const { completeAction } = useActions();
+  const { trackEvent } = useAnalyticsContext();
+  const { displayToast } = useToastNotification();
+  const { user, refetchBoot, showLogin } = useAuthContext();
+  const isMobile = !useMedia([mobileL.replace('@media ', '')], [true], false);
+  const isLaptopL = useMedia([laptopL.replace('@media ', '')], [true], false);
+  const {
+    mutateAsync: onAcceptMutation,
+    isLoading,
+    isSuccess,
+  } = useMutation(acceptFeatureInvitation, {
+    onSuccess: async () => {
+      await Promise.all([
+        completeAction(ActionType.AcceptedGenericReferral),
+        refetchBoot(),
+      ]);
+      router.push(redirectTo);
+    },
+    onError: (err: ApiErrorResult) => {
+      const message = err?.response?.errors?.[0]?.message;
+      displayToast(message ?? DEFAULT_ERROR);
+    },
+  });
+  const [isVideoOpen, setVideoOpen] = useState(false);
+
+  const onVideoModalClose = () => {
+    setVideoOpen(false);
+  };
+
+  const handleAcceptClick = () => {
+    const handleAccept = () =>
+      onAcceptMutation({
+        token,
+        referrerId: referringUser.id,
+        feature: ReferralCampaignKey.Generic,
+      });
+
+    if (!user) {
+      return showLogin({
+        trigger: AuthTriggers.GenericReferral,
+        options: {
+          onLoginSuccess: handleAccept,
+          onRegistrationSuccess: handleAccept,
+        },
+      });
+    }
+
+    // since in the page view, query params are tracked automatically,
+    // we don't need to send the params here explicitly
+    trackEvent({ event_name: AnalyticsEvent.AcceptInvitation });
+
+    return handleAccept();
+  };
+
+  useEffect(() => {
+    if (genericReferral === GenericReferral.Control) {
+      return;
+    }
+
+    router.push(redirectTo);
+    // router is an unstable dependency
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [redirectTo, genericReferral]);
+
+  return (
+    <div
+      style={{
+        background: `url(${cloudinary.referralCampaign.genericReferral.backgroundDark})`,
+        backgroundSize: 'cover',
+      }}
+      className="flex overflow-hidden relative flex-col laptop:flex-row flex-auto items-center p-2 laptop:p-6 h-[100vh]"
+    >
+      <span className="absolute top-8 left-1/2 laptop:left-12 -translate-x-1/2 laptop:translate-x-0">
+        <Logo showGreeting={false} position={LogoPosition.Relative} />
+      </span>
```

Comment:
I'm just wondering is this worth to become a component at the moment? it's quite small and Logo is already a component, so the only "big" thing we would be avoiding with this would be those classnames on the span. 
Anyways can change to a component if you think is better

Reply:
Ideally to me this whole page is a re-used template/layout as it has many common elements.
But agree with Lee, even though it's a simple wrapper would be great to have it as a common component.

Maybe you can even get away with simply a `classed` component?

Reply:
Ah, just pushed this in the common file at the moment, I believe that should be ok. But curiosity, @rebelchris why did you specifically suggested a `classed` component?

Reply:
Because it's a dumb component without logic, but anything will work.